guidelines/techpreview.md

```diff
@@ -0,0 +1,93 @@
+# What does it mean to be Tech Preview
+
+* You can change APIs that are clearly indicated as tech preview, without following a deprecating or backwards compatibility process.
+  * Note: you cannot change the type of an existing field w/o changing the apiversion.  Changing types has serialization 
+implications that can break existing code in significant ways.  Adding/removing/renaming fields is fine, if you need to change a
+type, rev the apiversion.  (You do not have to offer migration from the old api since it was tech preview).
+* You are not required to fix bugs customers uncover in your TP feature.  How much support you give to a customer who hits an 
+issue with the feature is up to you (but see below about doc+CEE training requirements).
+* You do not have to provide an upgrade path from a customer using your TP feature to the GA version of your feature.
+  * You must still support upgrading the cluster and your component, but it’s ok if the TP feature doesn’t work after the upgrade.
+  * If you can’t support upgrading the cluster+component when this TP feature is enabled, you must actively block the upgrade by 
+setting the upgradeable=false condition on your ClusterOperator and utilizing the TechPreviewNoUpgrade feature gate.
+* You still need to provide docs (which should make it clear the feature is tech preview)
+* You still need to provide education to CEE about the feature
+
+
+
+# Reasons to declare something Tech Preview
+
+* You aren’t confident you got the api right and want flexibility to change it without having to deal with migrations
+  * Bearing in mind the aforementioned restrictions on changing field types w/o revising the apiversion.
+* You aren’t confident in the implementation quality (scalability, stability, etc) and do not want to have to support customers using it in production in ways the implementation cannot handle
+
+
+# Downsides to declaring something Tech Preview
+* Since your feature requires special action to enable, you won’t get default CI coverage.  You may need to introduce your own 
+CI job that enables the TP feature if you want automated coverage
+* To date we have seen very few customers enabling feature gates (in part because they block upgrading that cluster) so if your
+feature is behind the cluster feature gate, you are unlikely to get meaningful feedback from the field to help you evolve the 
+feature anyway.  It may be better to just hold the feature until it’s GA ready.
+
+
+# Official process/mechanism for delivering a TP feature
+
+1. Your feature must be disabled by default
+1. Enabling it must require the admin enable a specific feature gate such as [TechPreviewNoUpgrade](https://docs.openshift.com/container-platform/4.1/nodes/clusters/nodes-cluster-enabling-features.html)
```

Comment:
Should we mention alternative mechanisms if the code is not part of the core platform (ie. an OLM operator).

Reply:
even if you're an olm operator you can require the cluster have TP enabled for your operator's feature to be enabled.

(or you can drop down to the unofficial processes below)